src/neo/SmartContract/Native/NeoToken.cs

```diff
@@ -75,14 +80,28 @@ internal override async ContractTask OnBalanceChanging(ApplicationEngine engine,
             CheckCandidate(engine.Snapshot, state.VoteTo, candidate);
         }
 
-        private async ContractTask DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccountState state)
+        private protected override async ContractTask PostTransfer(ApplicationEngine engine, UInt160 from, UInt160 to, BigInteger amount, StackItem data, bool callOnPayment)
+        {
+            await base.PostTransfer(engine, from, to, amount, data, callOnPayment);
+            var list = engine.CurrentContext.GetState<List<GasDistribution>>();
+            foreach (var distribution in list)
+                await GAS.Mint(engine, distribution.Account, distribution.Amount, callOnPayment);
```

Comment:
Correct if I'm wrong, but it shifts the order of NEO/GAS `onNEP17Payment` invocations, it was first GAS then NEO and now it'd be the other way around. How will contracts processing `onNEP17Payment` and backends processing notifications react to this change?

Reply:
We have verified the data on the mainnet and there is no change after applying this change.

Reply:
IMO that's not enough, there might be some assumptions of the order of events in backend code.

Reply:
Data is generated by contracts. If the data on the mainnet has not changed, then this proves that the change has not affected the logic of the contracts.